### ISSUE TITLE
[Bug]: [accessibility] [tagpicker] - Missing (Shift+)TAB functionality to select option from the list 

### ISSUE THREAD

**Issue by kolaps33** [microsoft/fluentui#31365]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
System:
    OS: Windows 10 10.0.22631
    CPU: (12) x64 Intel(R) Xeon(R) E-2176M  CPU @ 2.70GHz
    Memory: 16.36 GB / 31.74 GB
  Browsers:
    Edge: Chromium (123.0.2420.97)
    Internet Explorer: 11.0.22621.1
```

### Are you reporting Accessibility issue?

None

### Reproduction

https://stackblitz.com/edit/frj8eb?file=src%2Fexample.tsx

## Steps 
1. Navigate to the tagpicker
2. open tagpicker and navigate to any option in the list 
3. press TAB or Shift+TAB key

## Actual Behavior
nothing happen

## Expected Behavior
selected option from the list is chosen and inserted to tag picker

Why it should be implemented?
- this functionality works in V0 component https://fluentsite.z22.web.core.windows.net/0.66.2/components/dropdown/definition#types-multiple
- it is defined in the APG pattern: https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/
![image](https://github.com/microsoft/fluentui/assets/22620150/28e29225-8fac-4597-b435-30b85256175e)
- this functionality work with existing V9 combobox or dropdown component

### Requested priority

Blocking

### Are you willing to submit a PR to fix?

yes

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by bsunderhus** [comment]
keep open

**Comment by bsunderhus** [comment]
Expanding the scope of this issue: the expected Tab and Shift+Tab behavior should be provided consistently by all three related components: TagPicker, Dropdown, and Combobox.

When an option is highlighted in the open list, pressing Tab or Shift+Tab should select and commit that option before moving focus, matching the established behavior and the ARIA combobox pattern. This should be treated as a shared keyboard interaction requirement across these components, rather than a TagPicker-only issue.

**Comment by bsunderhus** [comment]
@kolaps33 I got a question related to the actual pattern, the example you're pointing towards is a single select combobox example. We have Dropdown, Combobox and TagPicker, all of them supporting single select and **multiselect**.

Combobox APG docs itself, it states clearly that an combobox is always single select

<img width="1447" height="568" alt="Image" src="https://github.com/user-attachments/assets/e1a19bca-6917-4a67-b990-6d9a0318517d" />

So how should Combobox, Dropdown and TagPicker should behaver regarding `Tab` and `Shift + Tab`  when on multiselect? should it be the same as single select? What would you expect in a multiselect? add a new selection? toggle the current selection to a new one?

**Comment by kolaps33** [comment]
Hi @bsunderhus 

from my point of view I would follow multiselect listbox example 
https://www.w3.org/WAI/ARIA/apg/patterns/listbox/examples/listbox-rearrangeable/

where in case of multiselect the TAB doesn't keep or do selection. 

So I would say that TAB in mutliselect option just close the listbox without invoking any action.

**Comment by bsunderhus** [comment]
Ok, then I suggest we update the "Expected Behavior" section on this bug description, this should be specific for single selection! Multiselect is not a problem here

**Comment by bsunderhus** [comment]
The actual codepath that does handle this use case is present:
https://github.com/microsoft/fluentui/blob/d27922755bebae866d9ffe86b7da44c27ec801ee/packages/react-components/react-combobox/library/src/utils/useTriggerSlot.ts#L198-L200

But it seems the `keydown` event is being swallowed on `Tab` interactions. By adding `tabster:movefocus` I can confirm that the event is being identified by tabster. We need to confirm if using `ignoreKeydown` for `Tab` is a valid solution for this case. @mshoho can you chime in here?!
